src/user/user.service.ts

```diff
@@ -15,19 +15,33 @@ export class UserService {
     return 'This action adds a new user';
   }
 
-  findAll() {
-    return this.userRepository.find();
+  async findAll(page, size): Promise<User[]> {
+    const skip = (page - 1) * size;
+    return this.userRepository.find({ skip, take: size });
   }
 
-  findOne(id: number) {
-    return `This action returns a #${id} user`;
-  }
+  // example to do: http://localhost:3000/api/user?searchBy=name&searchValue=Bob&page=1&size=1
+  async searchUsersByName(
+    searchBy: string,
+    searchValue: string,
+    page: number,
+    size: number,
+  ): Promise<User[]> {
+    const skip = (page - 1) * size;
+
+    const queryBuilder = this.userRepository
+      .createQueryBuilder('user')
+      .where(`user.${searchBy} LIKE :searchValue`, {
```

Comment:
you should **EVER NEVER** take a param from the user and use it directly in the database, I mean for the `searchBy` string, you only take the values and insert them into the database, not the field name
create an enum that converts from the param value to database column name, something like:
`
enum SearchBy{
  Name = "username",
  Email = "email",
}
`